### PR TITLE
[BUGFIX] Making an all-NULL column handling in RuleBasedProfiler more robust

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Insure that `ColumnPartition` metric implementation and `PartitionParameterBuilder` identify an all-`None` `numpy.ndarray` properly.
* Insure that heuristic data "sanitization" algorithm in top-level `ParameterBuilder` (e.g., converting `NaN` to `0.0`) handles an all-`None` `numpy.ndarray` properly.
* UAT using `SQL` and `Pandas` data in `Jupyter` notebook has been conducted satisfactorily.
* Existing tests pass.
* New specialized test for metric values "sanitization" added.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-465/GREAT-1230
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/expectations/metrics/column_aggregate_metrics/column_partition.py [hunk 1]
```diff
@@ -234,6 +234,9 @@ def _determine_bins_using_proper_units(
         # PRECISION NOTE: some implementations of quantiles could produce
         # varying levels of precision (e.g. a NUMERIC column producing
         # Decimal from a SQLAlchemy source, so we cast to float for numpy)
+        if min_ is None or max_ is None:
+            return None
```
Comment (Shinnnyshinshin) [inline great_expectations/expectations/metrics/column_aggregate_metrics/column_partition.py:238]: :heart:

**Review by Shinnnyshinshin — COMMENTED** [review]
Thank you very much for this work Alex. I'm just concerned about the UAT that was needed to confirm this bug + confirm that it was addressed. Is there a way that a unittest can be added to confirm the fix?

File: great_expectations/rule_based_profiler/attributed_resolved_metrics.py [hunk 2]
```diff
@@ -35,10 +35,10 @@ def _detect_illegal_array_type_or_shape(values: MetricValues) -> bool:
                     return True
 
                 # Components of different lengths cannot be packaged into "numpy.ndarray" type (due to undefined shape).
-                if all(isinstance(value, list) for value in values):
+                if all(isinstance(value, (list, tuple)) for value in values):
                     values_iterator: Iterator = iter(values)
                     first_value_length: int = len(next(values_iterator))
-                    current_value: List[Any]
+                    current_value: Sized[Any]
```
Comment (NathanFarmer) [inline great_expectations/rule_based_profiler/attributed_resolved_metrics.py:41]: What is `Sized`?

File: great_expectations/rule_based_profiler/attributed_resolved_metrics.py [hunk 3]
```diff
@@ -53,6 +53,14 @@ def _detect_illegal_array_type_or_shape(values: MetricValues) -> bool:
         return False
 
     if _detect_illegal_array_type_or_shape(values=metric_values):
+        value: MetricValue
+        if (
+            metric_values is not None
+            and isinstance(metric_values, (list, tuple))
```
Comment (NathanFarmer) [inline great_expectations/rule_based_profiler/attributed_resolved_metrics.py:59]: This should render #5927 moot, but for corner cases, is it possible we still want to also handle it downstream?

File: great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py [hunk 4]
```diff
@@ -573,43 +580,38 @@ def _sanitize_metric_computation(
...
-
-                        batch_metric_values.append(0.0)
                     else:
                         batch_metric_values.append(metric_value)
+                else:
+                    batch_metric_values.append(metric_value)
 
             metric_values_by_batch_id[batch_id] = batch_metric_values
 
         attributed_resolved_metrics.metric_values_by_batch_id = (
```
Comment (NathanFarmer) [inline great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py:613]: What happens with this variable if we no longer return it?

**Review by NathanFarmer — COMMENTED** [review]
I have a few questions!

File: great_expectations/rule_based_profiler/attributed_resolved_metrics.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/attributed_resolved_metrics.py:41]: @NathanFarmer It is a generic `Type` that admits the `len(.)` operation.  I found it to be helpful.  Thanks!

File: great_expectations/rule_based_profiler/attributed_resolved_metrics.py (hunk 3 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/attributed_resolved_metrics.py:59]: @NathanFarmer I am not sure -- I was hoping that once this is merged you would make best effort assessment and explain and re-issue the PR if you deem appropriate.  (Alternatively, we can reserve this higher-level check till the problem manifests itself vividly, making the development of an appropriate test a bit easier.). Thanks!

File: great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py (hunk 4 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py:613]: @NathanFarmer In this case, return is not needed; everything in the caller is "accepted as a side-effect" (the values are updated "in-situ").  Thanks!

File: tests/rule_based_profiler/parameter_builder/test_parameter_computations.py [hunk 5]
```diff
@@ -77,3 +89,168 @@ def test_bootstrap_point_estimate_efficacy(
...
+            "my_metric_6",
+            {
+                "batch_id_0": None,
+                "batch_id_1": "unparseable_metric_value_0",
+                "batch_id_2": "unparseable_metric_value_1",
+                "batch_id_3": None,
+            },
+        ),
+    ],
+)
```
Comment (Shinnnyshinshin) [inline tests/rule_based_profiler/parameter_builder/test_parameter_computations.py:193]: ❤️

**Review by Shinnnyshinshin — APPROVED** [review]
Looks good on my end ✅

**Review by NathanFarmer — APPROVED** [review]
Looks good to me as well!

**Pull request merged**